### PR TITLE
Add cmd

### PR DESCRIPTION
**Pull request by gbraad**
Added `cmd/vmproxy` and exposed the `SshServer` and `VncProxy`.

### CODE REVIEW

File: cmd/vmproxy/main.go [hunk 1]
```diff
@@ -10,20 +10,17 @@ import (
...
 
 func main() {
 	if len(os.Args) != 3 {
 		log.Fatal("usage: vmproxy <vm name> <vnc addrport>")
 	}
 
-	vmName = os.Args[1]
-	vncAddrPort = os.Args[2]
+	var vmName = os.Args[1]
+	var vncAddrPort = os.Args[2]
```
Comment (shayne) [inline cmd/vmproxy/main.go]:
```suggestion
	vmName := os.Args[1]
	vncAddrPort := os.Args[2]
```

File: cmd/vmproxy/main.go [hunk 2]
```diff
@@ -36,13 +33,13 @@ func main() {
 	var ln net.Listener
 
 	ln = must.Get(s.Listen("tcp", ":22"))
-	go sshServer(ln)
+	go vmproxy.SshServer(ln, vmName)
```
Comment (shayne) [inline cmd/vmproxy/main.go]: Capitalization of common initialisms

```suggestion
	go vmproxy.SSHServer(ln, vmName)
```

File: cmd/vmproxy/main.go [hunk 3]
```diff
@@ -36,13 +33,13 @@ func main() {
...
 
 	ln = must.Get(s.Listen("tcp", ":80"))
 	go promoteHTTPS(ln)
 
 	ln = must.Get(s.ListenTLS("tcp", ":443"))
-	vncProxy(s, ln)
+	vmproxy.VncProxy(s, ln, vncAddrPort)
```
Comment (shayne) [inline cmd/vmproxy/main.go]: Capitalization of common initialisms
```suggestion
	vmproxy.VNCProxy(s, ln, vncAddrPort)
```

File: ssh.go [hunk 4]
```diff
@@ -41,7 +45,8 @@ var (
 				Render
 )
 
-func sshServer(ln net.Listener) {
+func SshServer(ln net.Listener, vm string) {
```
Comment (shayne) [inline ssh.go]: Capitalization of common initialisms
```suggestion
func SSHServer(ln net.Listener, vm string) {
```

File: vnc.go [hunk 5]
```diff
@@ -17,7 +17,7 @@ import (
 //go:embed noVNC
 var f embed.FS
 
-func vncProxy(s *tsnet.Server, tlsLn net.Listener) {
+func VncProxy(s *tsnet.Server, tlsLn net.Listener, vncAddrPort string) {
```
Comment (shayne) [inline vnc.go]:
```suggestion
func VNCProxy(s *tsnet.Server, tlsLn net.Listener, vncAddrPort string) {
```

File: ssh.go [hunk 6]
```diff
@@ -41,7 +45,8 @@ var (
...
+	vmName = vm
```
Comment (shayne) [inline ssh.go:48]:
```suggestion
	# TODO(shayne): Don't use a package global.
	vmName = vm
```

File: ssh.go [hunk 7]
```diff
@@ -21,6 +21,10 @@ import (
 	lm "github.com/charmbracelet/wish/logging"
 )
 
+var (
+	vmName string
+)
+
```
Comment (shayne) [inline ssh.go]:
```suggestion
var vmName string
```

File: README.md [hunk 8]
```diff
@@ -1,17 +1,30 @@
 # vmproxy
 Tailscale Proxy for VNC and libvirt virsh control over a VM
 
+
```
Comment (shayne) [inline README.md]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -1,17 +1,30 @@
...
+## Building and running
+To build from source and run in dev mode:
+
+```
+go run ./cmd/vmproxy <vm name> <vnc addrport>
+```
+
+Note: requires go 1.20 to compile
```
Comment (shayne) [inline README.md]:
```suggestion
Note: Requires Go 1.20
```

**Review by shayne — CHANGES_REQUESTED** [review]

File: ssh.go (hunk 6 above)
Comment (gbraad) [reply ssh.go:48]: This was one of my biggest itches with doing this, but a quick fix until a better solution is in place. At the moment, the code wasn't expecting a module.

**Comment by gbraad** [comment]
Suggestions make sense. Will push an update soon(-ish).

File: ssh.go (hunk 6 above)
Comment (gbraad) [reply ssh.go:48]:
```
// TODO(shayne): Don't use a package global.
```

;-)

**Comment by gbraad** [comment]
PTAL

File: README.md [hunk 10]
```diff
@@ -8,10 +8,24 @@ Resume.
...
+## Building and running
+To build from source and run in dev mode:
+
+```
+go run ./cmd/vmproxy <vm name> <vnc addrport>
+```
+
+For the initial run you need to register with `TS_AUTHKEY`.
+
+Note: requires Go 1.20
```
Comment (shayne) [inline README.md]:
```suggestion
Note: Requires Go 1.20
```

Comment (gbraad) [reply README.md]: oh darn... missed that 🤦

Comment (gbraad) [reply README.md]: Punctuation rules as I know didn't capitalize unless it was followed by a full sentence. Here is more of a statement, 'that is to say'.

**Comment by gbraad** [comment]
Repushed.

**Review by shayne — APPROVED** [review]

**Pull request merged**